### DISCUSSION TITLE
How to install this without conflicts with unity newton soft?

### DISCUSSION THREAD

**Discussion by mls-icaro** [applejag/Newtonsoft.Json-for-Unity#135] [Q&A] [answered]
I am trying install but I have conflicts with actual newtonsoft installed inside Unity.

 'Packages/jillejr.newtonsoft.json-for-unity/Plugins/Newtonsoft.Json AOT/Newtonsoft.Json.dll' conflicts with:
  'Packages/com.unity.nuget.newtonsoft-json/Runtime/AOT/Newtonsoft.Json.dll' (current owner)
We can't assign a new GUID because the asset is in an immutable folder. The asset will be ignored.

I a using Unity 2020.3.11f

**Comment by mls-icaro** [comment]
I Go to Advanced Package Managers settings and check the show all dependencies package. I remove now the Collab package that has as dependency the newtonsoft package from Unity. After that do a Reimport All.

JilleJr newtonsoft installed!

**Reply by applejag** [reply]
Great that you found a solution around it!

**Comment by applejag** [comment, marked as answer]
Sorry to say that you cannot have them side by side. The reason for the GUID collision is that Unity just forked my repository, but didn't refresh the GUIDs.

But even if the GUIDs weren't colliding, there would still be two Newtonsoft.Json assemblies causing collisions on that level.

There are some other things as well that are in the way for this to succeed, so the short answer is no, you cannot have both my package and the "official"  Unity package as well.

We're in the midst of moving my package over to the UnityTechnologies organization here on GitHub, but those plans currently looks like will be accomplished in 2022 as soonest.

Thanks for your patience!

**Comment by mls-icaro** [comment]
After I change to you newtonsoft package, I get the error in apk:

`sharedassets0.assets' is corrupted! Remove it and launch unity again!`

Now, when I back to normal newtonsoft package, the error does not gone. It is persists.

**Reply by applejag** [reply]
When stuff gets corrupt I tend to close Unity, delete the Library folder (it's all autogenerated stuff anyway) and boot up the project again, letting unity regenerate all files correctly.

Could you try that?

**Reply by mls-icaro** [reply]
I try one time. So I back to my commit that working the apk building without JillJr newtonsoft.

I will try agan with JilleJr in this weekend and I post here the result.

**Reply by mls-icaro** [reply]
Yes, this worked. Thanks.

**Comment by BenWoodford** [comment]
Does anyone know if the Unity one can be included as a dependency in 2019 and 2020? I notice it only shows in 2021 because of Collab...

**Reply by applejag** [reply]
You can follow the guide here: https://github.com/jilleJr/Newtonsoft.Json-for-Unity/wiki/Install-official-via-UPM

In case you still can't add the package by name via the package manager window, then try the "via manifest.json" path at the bottom.
